### PR TITLE
Map and gallery template added to tags page

### LINKED ISSUE #807: Tags display with "tour"
The new front page is awesome: http://mapknitter-stable.laboratoriopublico.org/

We're looking forward to adding such a "top map" as shown below, to the tags page, for example: https://mapknitter.org/tags/featured

image

We could combine the new gallery template (http://mapknitter-stable.laboratoriopublico.org/gallery) with the a top map, and what would be even cooler is if we had a button that offered a "tour" that uses `map.flyTo(...)` to cycle through the different maps on the page. (we'd have to calculate a bounding box perhaps to zoom to, or something, based on each map's images).

This isn't top priority but the basic addition of a map doesn't look like it'd be too hard; the "fly around" tour could come later!

### PR DESCRIPTION
**Pull request by divyabaid16**
Fixes #807  (<=== Add issue number here)

Selection_202

* [x] PR is descriptively titled 📑 and links the original issue above 🔗
* [x] tests pass -- look for a green checkbox ✔️ a few minutes after opening your PR -- or run tests locally with `rake test`
* [x] code is in uniquely-named feature branch and has no merge conflicts 📁
* [x] screenshots/GIFs are attached 📎 in case of UI updation
* [x] ask `@publiclab/mapknitter-reviewers` for help, in a comment below

Thanks!

### CODE REVIEW

**Comment by divyabaid16** [comment]
@publiclab/mapknitter-reviewers

**Comment by jywarren** [comment]
Oh wow, this is awesome! I was thinking ideally we might show only maps
that are tagged -- @sagarpreet-chadha, would it be possible to pass a new
RSS URL into the layer, so it'd otherwise be identical but would display
https://mapknitter.org/feeds/tag/featured or
https://mapknitter.org/feeds/tag/barataria instead of the default?

**Comment by divyabaid16** [comment]
I'm not getting this. @ananyaarun would you like to give some input on this?
Here, we have to show only the maps that are **tagged** or has some tag associated with it.
Thanks!

**Comment by ananyaarun** [comment]
Hey @divyabaid16 
What is the exact issue you are facing ?

**Comment by divyabaid16** [comment]
@ananyaarun have a look at https://github.com/publiclab/mapknitter/pull/821#issuecomment-509658017 comment. 

The page that displays this is 
https://mapknitter.org/tags/featured

**Comment by ananyaarun** [comment]
Hi @divyabaid16 
I am completely new to mapknitter I had a few questions , So in the https://mapknitter.org/tags/featured page I see on top the mapknitter layer is displayed (this is the LEL mapknitter layer)
But the maps ie the ones with featured tags displayed below have nothing to do with LEL layers do they ?

**Comment by divyabaid16** [comment]
@jywarren just to clear, were you asking me to sum up the coordinates of all the maps in LEL, find their average and then zoom in to that point in the top map?
And show that point in LEL mapknitter layer?

**Comment by jywarren** [comment]
Ah actually all the maps that are tagged with the given tag. But that'll be
all maps if we adapt the Json endpoint to allow a tag parameter!

For the short term fix, you can just use leaflet's getBounds feature to
pass all coords in and it'll calculate a bounds to include them all. How
does that sound? Thanks!!

**Comment by jywarren** [comment]
Hi all! How is this coming along -- can I help out? Thanks!!!

**Comment by divyabaid16** [comment]
Ya, I'm trying. Sorry, It took me a bit longer.

**Comment by jywarren** [comment]
no problem, and if you get stuck, ping me! Thanks, Divya!!!

File: app/views/tags/index.html.erb [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+<%= stylesheet_link_tag "https://api.mapbox.com/mapbox.js/plugins/leaflet-fullscreen/v1.0.1/leaflet.fullscreen.css" %>
+
+<%= javascript_include_tag('/lib/leaflet-spin/example/spin/dist/spin.min.js') %>
+<%= javascript_include_tag('/lib/leaflet-spin/example/leaflet.spin.min.js') %>
+
+<div class="container-fluid">
+  <div id="mapknitter-unique" class="mx-auto" style="height: 450px; width: 100%;">
+  </div>
+  <div class="galary-template">
+    <%= render :template => 'front_ui/gallery' %>
```
Comment (cesswairimu) [inline app/views/tags/index.html.erb]: @divyabaid16 My suggestion is here we  render the maps partial like so
`  <%= render :partial => 'front_ui/maps', :locals => { :maps => @maps } %>`
So we don't include the other tabs in gallery i.e `All Maps`  `anonymous` etc

File: app/views/tags/index.html.erb [hunk 2]
```diff
@@ -0,0 +1,129 @@
...
+<%= javascript_include_tag('/lib/leaflet-fullhash/leaflet-fullHash.js') %>
+<%= javascript_include_tag('https://api.mapbox.com/mapbox.js/plugins/leaflet-fullscreen/v1.0.1/Leaflet.fullscreen.min.js') %>
+<%= stylesheet_link_tag "https://api.mapbox.com/mapbox.js/plugins/leaflet-fullscreen/v1.0.1/leaflet.fullscreen.css" %>
+
+<%= javascript_include_tag('/lib/leaflet-spin/example/spin/dist/spin.min.js') %>
+<%= javascript_include_tag('/lib/leaflet-spin/example/leaflet.spin.min.js') %>
+
+<div class="container-fluid">
+  <div id="mapknitter-unique" class="mx-auto" style="height: 450px; width: 100%;">
+  </div>
```
Comment (cesswairimu) [inline app/views/tags/index.html.erb]: Maybe we can add a heading here that say we are displaying the "maps with the tag tag_name". as a title. Or have a little text box as plots2 does it here  https://publiclab.org/tag/water-quality. What do you think?

**Comment by divyabaid16** [comment]
I have made the changes @cesswairimu.
Thanks!

File: app/views/tags/index.html.erb [hunk 3]
```diff
@@ -0,0 +1,111 @@
...
+  <div class="galary-template">
+    <br> <br>
+    <h2 class="section-title text-center"> <%= @title %></h2>
+    <hr style="max-width: 500px;">
+    <br> <br>
+    <%= render :partial => 'front_ui/maps', :locals => { :maps => @maps } %>
```
Comment (jywarren) [inline app/views/tags/index.html.erb]: Nice!!!

**Comment by jywarren** [comment]
This looks great. merging - and shall we think of how to show people how to use this feature? Like, should this page have a section that says `use the tag _____ to add a map to this collection` or on a map view page, an area for inputting tags that explains this too? We could move the tags section from the map show page to the view only map page that @cesswairimu built in #777 ? What do you think?

**Comment by jywarren** [comment]
Oops, closed by mistake!!! I'll merge in any case, since it already passed. Thanks!

**Pull request merged**